topics/wallet-interaction-standard.md

```diff
@@ -0,0 +1,265 @@
+        },
+        "challenge": "3Q/X4Vim3kAhAiLzdHi8h/AC1nzpDyfT4D/6LpxFIkc=",
+        "signature": "rEm2+pM0LdlWP3Rvyry1uD9YcLUGI4eaxENqYPcTaP8Wj12v7SEIWX/fokcy99XnPQIo879J0+7HP1I1UgDB3w=="
+    }
+}
+```
+
+### `canister_call`
+
+Once the connection between the relying party and the wallet is established, and the relying party has been granted the `canister_call` permission scope, the relying party can request the wallet to execute canister calls.
+
+#### Types
+
+- `text`: A plain `string` value.
+- `blob`: A `string` value describing binary data encoded in base64.
+- `nat`: A `string` value of an unsigned 64-bit integer.
+
+#### Request
+
+`version` (`text`): The version of the standard used.
+
+`network`: Network details on which the call should be executed.
+- `chainId` (`text`) - The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`canisterId` (`text`): The id of the canister on which the call should be executed.
+
+`sender` (`text`): The principal requested to execute the call.
+
+`method` (`text`): The name of the call method to be executed.
+
+`arg` (`blob`): The arguments for the call.
+
+
+#### Response
+
+`version` (`text`): The version of the standard used.
+
+`network`: Network details on which the call was executed.
+- `chainId` (`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`contentMap`: The actual request content as specified [here](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-call).
+- `request_type` (`text`)
+- `sender` (`blob`)
+- `nonce` (`blob`, optional)
+- `ingress_expiry` (`nat`)
+- `canister_id` (`blob`)
+- `method_name` (`text`)
+- `arg` (`blob`)
+
+`certificate` (`blob`): The certificate returned by the `read_state` call as specified [here](https://internetcomputer.org/docs/current/references/ic-interface-spec/#certificate). The value is CBOR-encoded.
+
+#### Use-Case
+
+1. The relying party sends a `canister_call` request to the wallet.
+2. Upon receiving the request, the wallet validates whether the relying party has the permission to request the action and processes the message following the [ICRC-21]() specification.
+    - If the user approves the request:
+        - The wallet sends the call to the IC, retrieves its content map ([call](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-call), [query](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-query)) and [calculates a request id](https://internetcomputer.org/docs/current/references/ic-interface-spec/#request-id) based on it.
```

Comment:
Query calls work differently than update calls (i.e. they are synchronous, thus require no polling, but the result is also not certified).
Given that adding certification to query calls is application specific I would _not_ support signing `query` calls using the wallet.

Rather, it is possible to send messages to canisters as `update` even for canister methods marked as `query` in which case the call is executed in a replicated fashion and the result is certified as well (and verification works the same as for regular `update` calls).

So I would just add a note here that all calls (including queries) are sent as `update` calls.